include/exadg/incompressible_navier_stokes/preconditioners/multigrid_preconditioner_momentum.cpp

```diff
@@ -142,8 +142,23 @@ MultigridPreconditioner<dim, Number>::fill_matrix_free_data(
 
   matrix_free_data.insert_dof_handler(&(*this->dof_handlers[level]), "std_dof_handler");
   matrix_free_data.insert_constraint(&(*this->constraints[level]), "std_dof_handler");
-  matrix_free_data.insert_quadrature(dealii::QGauss<1>(this->level_info[level].degree() + 1),
-                                     "std_quadrature");
+  if(this->dof_handlers[level]->get_triangulation().all_reference_cells_are_hyper_cube())
+  {
+    matrix_free_data.insert_quadrature(dealii::QGauss<1>(this->level_info[level].degree() + 1),
+                                       "std_quadrature");
+  }
+  else if(this->dof_handlers[level]->get_triangulation().all_reference_cells_are_simplex())
+  {
+    matrix_free_data.insert_quadrature(dealii::QGaussSimplex<dim>(this->level_info[level].degree() +
+                                                                  1),
+                                       "std_quadrature");
+  }
+  else
+  {
+    AssertThrow(false,
+                dealii::ExcMessage("Only pure hypercube or pure simplex meshes are implemented for "
+                                   "IncNS::Multigrid."));
+  }
```

Comment:
In `IncNS::SpatialOperatorBase`, you have used `get_gauss_type_quadrature<dim>()`, which seems to handle both hypercube and simplex elements. Could you comment why you did not use this here?

Reply:
There was no reason. I changed the other occurrence of `matrix_free_data.insert_quadrature(...)` to this `if-else` logic as well, to keep the design consistent.

Reply:
What is `IncNS::Multigrid`?

The `get_gauss_type_quadrature<dim>()` is actually more elegant in my opinion, so I will open an issue so that we have a reminder to change this in the future.

Reply:
see issue #430